test/code/plugins/agent_topology_request_script_plugintest.rb

```diff
@@ -229,15 +210,21 @@ def test_valid_data_empty_process_stats
     atr.EntityTypeId = @entityTypeId
     atr.AuthenticationCertificate = @authcert
 
-    IO.write(OS_INFO, "OSName=Ubuntu\n OSManufacturer=Canonical Group Limited\n OSVersion=14.04")
+    IO.write(OS_INFO, VALID_OS_INFO)
     IO.write(CONF_OMSADMIN, "This is a test omsadmin.conf file")
+    IO.write(PID_FILE, CONSTANT_PID)
 
-    atr.get_telemetry_data(OS_INFO, CONF_OMSADMIN)
+    atr.get_telemetry_data(OS_INFO, CONF_OMSADMIN, PID_FILE)
+    verify_complete_xml(atr, expect_telemetry = false)
+ 
+    flexmock(AgentTopologyRequest).flexmock_teardown
+  end    
 
+  def verify_complete_xml(atr, expect_telemetry = true)
```

Comment:
Git's diff has parsed this a little strangely; just know that the new file's lines 223-271 are a single new helper method called verify_complete_xml